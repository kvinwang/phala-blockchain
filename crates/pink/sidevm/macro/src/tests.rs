#[test]
fn test_ocall() {
    let stream = crate::macro_ocall::patch(syn::parse_quote! {
        pub trait Ocall {
            #[ocall(id = 101)]
            fn call_slow(a: i32, b: i32) -> i32;

            #[ocall(id = 103, fast_input)]
            fn call_fi(a: i32, b: i32) -> i32;

            #[ocall(id = 104, fast_return)]
            fn call_fo(a: i32, b: i32) -> i32;

            #[ocall(id = 102, fast_input, fast_return)]
            fn poll_fi_fo(a: i32, b: i32) -> i32;
        }
    });
    insta::assert_display_snapshot!(rustfmt_snippet::rustfmt_token_stream(&stream).unwrap())
}

#[test]
fn test_main() {
    let stream = crate::macro_main::patch(syn::parse_quote! {
        async fn the_main() {
            sleep(1).await
        }
    });
    insta::assert_display_snapshot!(rustfmt_snippet::rustfmt_token_stream(&stream).unwrap())
}
