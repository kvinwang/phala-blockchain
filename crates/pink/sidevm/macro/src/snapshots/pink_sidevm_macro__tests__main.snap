---
source: crates/pink/sidevm/macro/src/tests.rs
assertion_line: 28
expression: "rustfmt_snippet::rustfmt_token_stream(&stream).unwrap()"
---
#[no_mangle]
extern "C" fn sidevm_poll() -> i32 {
    use pink_sidevm_env::{poll_with_dummy_context, reexports::once_cell::sync::Lazy};
    use std::{future::Future, pin::Pin, sync::Mutex, task::Poll};
    async fn the_main() {
        sleep(1).await
    }
    static MAIN_FUTURE: Lazy<Mutex<Pin<Box<dyn Future<Output = ()> + Sync + Send>>>> =
        Lazy::new(|| Mutex::new(Box::pin(the_main())));
    match poll_with_dummy_context(MAIN_FUTURE.lock().unwrap().as_mut()) {
        Poll::Ready(()) => 1,
        Poll::Pending => 0,
    }
}

